src/UserService.Broker/Consumers/UserLoginConsumer.cs

```diff
@@ -36,30 +42,22 @@ private object GetUserCredentials(IUserCredentialsRequest request)
         {
             DbUserCredentials dbUserCredentials;
 
-            if (!IsEmail(request.LoginData))
+            try
             {
-                dbUserCredentials = _credentialsRepository.GetUserCredentialsByLogin(request.LoginData);
-
-                if (dbUserCredentials == null)
+                if (!IsEmail(request.LoginData))
                 {
-                    throw new NotFoundException($"User credentials for user with login: '{request.LoginData}' was not found.");
+                    dbUserCredentials = GetDbUserCredentialsByLogin(request.LoginData);
                 }
-            }
-            else
-            {
-                var dbUser = _userRepository.GetUserByEmail(request.LoginData);
-
-                if (dbUser == null)
+                else
                 {
-                    throw new NotFoundException($"User with email: '{request.LoginData}' was not found.");
+                    dbUserCredentials = GetDbUserCredentialsByUserId(request.LoginData);
                 }
+            }
+            catch (Exception exception)
+            {
+                _logger.LogError(exception, exception.Message);
 
-                dbUserCredentials = _credentialsRepository.GetUserCredentialsByUserId(dbUser.Id);
-
-                if (dbUserCredentials == null)
-                {
-                    throw new NotFoundException($"User credentials for user with email: '{request.LoginData}' was not found.");
-                }
+                throw new Exception(exception.Message);
```

Comment:
`throw new Exception(exception.Message);` - it's bad

Reply:
When we throw an error, it is written to the logger, so we don't have to do it.

Reply:
Reworked